classes/Context.php

```diff
@@ -422,49 +422,15 @@ public function getTranslatorFromLocale($locale)
             (new Filesystem())->remove($cache_file);
         }
 
-        $adminContext = defined('_PS_ADMIN_DIR_');
-        $translator->addLoader('xlf', new XliffFileLoader());
-
-        $sqlTranslationLoader = new SqlTranslationLoader();
-        if (null !== $this->shop) {
-            $sqlTranslationLoader->setTheme($this->shop->theme);
-        }
-
-        $translator->addLoader('db', $sqlTranslationLoader);
-        $notName = $adminContext ? '^Shop*' : '^Admin*';
-
-        $finder = Finder::create()
-            ->files()
-            ->name('*.' . $locale . '.xlf')
-            ->notName($notName)
-            ->in($this->getTranslationResourcesDirectories());
+        // only way to reset the catalogue
+        $translator->addLoader('Array', new ArrayLoader());
+        $translator->addResource('Array', [], $locale);
 
-        foreach ($finder as $file) {
-            list($domain, $locale, $format) = explode('.', $file->getBasename(), 3);
-
-            $translator->addResource($format, $file, $locale, $domain);
-            if (!$this->language instanceof PrestashopBundle\Install\Language) {
-                $translator->addResource('db', $domain . '.' . $locale . '.db', $locale, $domain);
-            }
-        }
+        $adminContext = defined('_PS_ADMIN_DIR_');
+        $withDB = !$this->language instanceof PrestashopBundle\Install\Language;
```

Comment:
I don't understand this check? In which case don't you want the database to be loaded? Maybe it's worth a comment here

Besides what other type can `$this->language` be? `\Language`? So it means you only disable the db loading during install process? Why?

Reply:
We don't need to load DB translations when `$this->language` in an instance of `PrestashopBundle\Install\Language` because that's only translations for the web installer and when we load the installer language we didn't configured the DB yet...

Reply:
oooookay, this makes sense but maybe you can add a comment explaining that, because it's not that intuitive at first ^^

Reply:
🆙